Address issue #396

xhoenner opened the pull request:
Made Tag Expected Life Time in tag creation and edit pages mandatory. Could you please review this PR @bhasin85?

bhasin85 commented:
@xhoenner Trigger the build again.... It should pass all the checks.

bhasin85 commented on grails-app/domain/au/org/emii/aatams/Tag.groovy:
```diff
@@ -29,7 +29,7 @@ class Tag extends Device implements Embargoable {
         project(nullable:true)
-        expectedLifeTimeDays(nullable:true)
+        expectedLifeTimeDays(nullable:false)
         sensors (validator: { sensors ->
```
It will be good to add few unit tests checking the following scenarios.

1. Tag creation pass, when expectedLifeTimeDays value is passed
1. Tag creation fails when expectedLifeTimeDays value is not passed
1. Tag creation fails when expectedLifeTimeDays value is 2147483648 or greater
1. Tag creation fails when expectedLifeTimeDays value is (-2147483648 or less) or (less than 0)

bhasin85 commented:
Can you please update the branch again with Master.... Thanks :+1:

xhoenner commented:
@bhasin85 I fixed up existing issues with unit tests and added in the tests you recommended. Thanks heaps for your kind help!

Similarly to https://github.com/aodn/aatams/pull/395, the Travis build failed due to an issue with postgis. The log also says:
```bash
The command "./init_test_db.sh" failed and exited with 1 during .
```

xhoenner commented:
Both PRs have now passed all unit and integration tests @bhasin85 so it'd be great if you could review those in the next few days.

bhasin85 commented:
@xhoenner  This one looks good. Thanks for fixing it up. :+1: 

![giphy](https://user-images.githubusercontent.com/2552378/30526973-974885c2-9c66-11e7-8f73-d87db36f25ea.gif)

xhoenner commented:
I never thought it'd be this painful to go through all those tests

![giphy](http://www.reactiongifs.com/wp-content/uploads/2012/11/crying-man.gif)
